core/templates/pages/teach-page/teach-page.component.ts

```diff
@@ -15,64 +15,153 @@
-
+  isWindowNarrow: boolean = false;
+  classroomUrlFragment: string;
+  classroomUrl :string;
+  displayedTestimonialId: number;
+  testimonialCount: number;
+  testimonials = [];
+  userIsLoggedIn: boolean = null;
   constructor(
     private siteAnalyticsService: SiteAnalyticsService,
     private urlInterpolationService: UrlInterpolationService,
-    private windowRef: WindowRef
+    private windowDimensionService: WindowDimensionsService,
+    private windowRef: WindowRef,
+    private userService: UserService,
+    private loaderService: LoaderService
   ) {}
 
   ngOnInit(): void {
-    const hash = this.windowRef.nativeWindow.location.hash.slice(1);
-    if (this.ALLOWED_TABS.includes(hash)) {
-      this.activeTabName = hash;
-    }
-    this.windowRef.nativeWindow.onhashchange = () => {
-      const hashChange = this.windowRef.nativeWindow.location.hash.slice(1);
-      if (this.ALLOWED_TABS.includes(hashChange)) {
-        this.activeTabName = hashChange;
-      }
-    };
+    this.isWindowNarrow = this.windowDimensionService.isWindowNarrow();
+    this.displayedTestimonialId = 0;
+    this.testimonialCount = 4;
+    this.testimonials = this.getTestimonials();
+    this.classroomUrl = this.urlInterpolationService.interpolateUrl(
+      '/learn/<classroomUrlFragment>', {
+        classroomUrlFragment: splashConstants.DEFAULT_CLASSROOM_URL_FRAGMENT
+      });
+    this.loaderService.showLoadingScreen('Loading');
+    this.userService.getUserInfoAsync().then((userInfo) => {
+      this.userIsLoggedIn = userInfo.isLoggedIn();
+      this.loaderService.hideLoadingScreen();
+    });
   }
 
-  onTabClick(tabName: string): Window {
-    // Update URL hash.
-    this.windowRef.nativeWindow.location.hash = '#' + tabName;
-    this.activeTabName = tabName;
-    return this.windowRef.nativeWindow;
+  // The 2 functions below are to cycle between values:
+  // 0 to (testimonialCount - 1) for displayedTestimonialId.
+  incrementDisplayedTestimonialId(): void {
+  // This makes sure that incrementing from (testimonialCount - 1)
+  // returns 0 instead of testimonialCount,since we want the testimonials
+  // to cycle through.
+    this.displayedTestimonialId = (
+      this.displayedTestimonialId + 1) % this.testimonialCount;
   }
 
-  getStaticImageUrl(imagePath: string): string {
-    return this.urlInterpolationService.getStaticImageUrl(imagePath);
+  decrementDisplayedTestimonialId(): void {
+  // This makes sure that decrementing from 0, returns
+  // (testimonialCount - 1) instead of -1, since we want the testimonials
+  // to cycle through.
+    this.displayedTestimonialId = (
+      this.displayedTestimonialId + this.testimonialCount - 1) %
+      this.testimonialCount;
+  }
+
+  getTestimonials():[{}, {}, {}, {}] {
+    return [{
+      quote: 'I18N_TEACH_TESTIMONIAL_1',
+      studentDetails: 'I18N_TEACH_STUDENT_DETAILS_1',
+      imageUrl: '/splash/mira.png',
+      imageUrlWebp: '/splash/mira.webp',
```

Comment:
Should we unify the naming of these images, for example only use the name in a lowercase letters?

Reply:
done.